gammapy/image/mask.py

```diff
@@ -95,8 +94,38 @@ def plot(self, ax=None, fig=None, **kwargs):
 
     @lazyproperty
     def distance_image(self):
-        """Map containting the distance to the nearest exclusion region."""
-        return exclusion_distance(self.mask)
+        """Distance to nearest exclusion region.
+
+        Compute distance map, i.e. the Euclidean (=Cartesian 2D)
+        distance (in pixels) to the nearest exclusion region.
+
+        We need to call distance_transform_edt twice because it only computes
+        dist for pixels outside exclusion regions, so to get the
+        distances for pixels inside we call it on the inverted mask
+        and then combine both distance images into one, using negative
+        distances (note the minus sign) for pixels inside exclusion regions.
+
+        If data consist only of ones, it'll be supposed  to be far away from zero pixsels,
+        so in capacity of answer it should be return the matrix with the shape as like as
+        data but packed by constant value Max_Value (MAX_VALUE = 1e10).
+
+        Returns
+        -------
+        distance : `~gammapy.image.SkyMap`
+            Sky map of distance to nearest exclusion region.
+        """
+        if np.all(self.mask == 1):
```

Comment:
I know it's a matter of taste, but I would suggest to have the special case return the special result, and then to have the normal case not be indented in an else block.

I.e. use this:
```python
if np.all(self.mask == 1):
    return SkyMap.empty_like(self, fill=MAX_VALUE)

# continue here with what you have in the else block, but not indented.
```